cosmic-applet-network/src/app.rs

```diff
@@ -229,96 +350,304 @@ impl Application for CosmicNetworkApplet {
+                                        }))
+                                        .width(Length::Units(24))
+                                        .height(Length::Units(24)),
+                                ]
+                                .spacing(8)
+                                .padding([0, 24]);
+                                content = content.push(connecting);
+                            }
+                            NewConnectionState::Failure(access_point) => {
+                                let id = row![
+                                    icon("network-wireless-symbolic", 24)
+                                        .style(Svg::Custom(|theme| svg::Appearance {
+                                            color: Some(theme.palette().text),
+                                        }))
+                                        .width(Length::Units(24))
+                                        .height(Length::Units(24)),
+                                    text(&access_point.ssid).size(14),
+                                ]
+                                .align_items(Alignment::Center)
+                                .width(Length::Fill)
+                                .padding([0, 24])
+                                .spacing(12);
+                                content = content.push(id);
+                                let col = column![
+                                    text(fl!("unable-to-connect")),
+                                    text(fl!("check-wifi-connection")),
+                                    row![
+                                        button(Button::Secondary)
+                                            .custom(vec![container(text("Cancel"))
+                                                .padding([0, 24])
+                                                .into()])
+                                            .on_press(Message::CancelNewConnection),
+                                        button(Button::Secondary)
+                                            .custom(vec![container(text("Connect"))
+                                                .padding([0, 24])
+                                                .into()])
+                                            .on_press(Message::SelectWirelessAccessPoint(
+                                                access_point.clone()
+                                            ))
+                                    ]
+                                    .spacing(24)
+                                ]
+                                .spacing(16)
+                                .padding([0, 48])
+                                .align_items(Alignment::Center);
+                                content = content.push(col);
+                            }
+                        }
+                    } else if self.nm_state.wifi_enabled {
+                        let mut list_col = column![];
```

Comment:
Here you know the length of access points to push `self.nm_state.wierless_access_points`
Pushing one by one is several heap allocations. Create a vec with capacity of `sel.nm_state.wierless_access_points.len()`, then push into that vec, then at the end create the column with `Column::with_children`, then it is only one heap allocation.

Reply:
I believe you do this in a few spots